core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/GraalVM.java

```diff
@@ -200,17 +197,20 @@ public int compareTo(Version o) {
             return this.version.compareTo(o.version);
         }
 
-        static Version of(Stream<String> lines) {
-            List<String> linesList = toList(lines); // relies on ordering of the stream
-            if (linesList.size() == 3) {
+        static Version of(Stream<String> output) {
+            List<String> lines = output
+                    .dropWhile(l -> !l.startsWith("GraalVM") && !l.startsWith("native-image"))
```

Comment:
@gsmet How is this not dropping lines of the form (third line in the new output):
```
Substrate VM GraalVM CE 20.0.1+9.1 (build 20.0.1+9, serial gc)
```
and
```
OpenJDK Runtime Environment Mandrel-23.0.0.0-Final (build 17.0.7+7)
```

What am I missing?

Reply:
I think the safer option is to pull the image before running `--version` on it.

Reply:
@jerboaa it's `dropWhile`: it's dropping the lines while the condition is true, then it keeps all the lines.

Reply:
As for changing entirely the way it is done, I'm not comfortable to do it now but could be convinced for a future version.

Reply:
FWIW, I discovered this method today while implementing this :).

Reply:
@gsmet Gotcha. Fair enough then.